### ISSUE TITLE
configure highlighting of Literal Strings

### ISSUE THREAD

**Issue by indika-dev** [rebelot/kanagawa.nvim#48]
Hi!

I'm currently implementing a grammar with [LuLPEG](https://github.com/pygy/LuLPeg) and that grammar is defined inside a 1st-Level Literal String. Literal Strings are now highlighted by kanagawa.vim as you can see in the screenshot. Unfortunately, the whole block got repainted if I move the cursor a line up or down, which is very annoying for me. I tried to redefine the colors and highlighting groups as described for this color scheme , but I hadn't any luck.
So, what is the correct configuration for no highlighting of Literal Strings?

![image](https://user-images.githubusercontent.com/56093026/172821252-f49a00bc-a9dc-48e8-bb99-94291e784c57.png)

If you want to try that by your yourself, this is the code I used in the screenshot:
```
local mail_pat = [=[
    groups            <- (group (%s* ([,;] %s*)+ group)*) -> {}
    group             <- (
                            {:name: <phrase> :} %s* <addr>                  /
                            {:name: <uq_phrase> :} %s* "<" <addr_spec> ">"  /
                            <addr> %s* {:name: <phrase> :}                  /
                            "<" <addr_spec> ">" %s* {:name: <uq_phrase> :}  /
                            <addr>                                          /
                            {:name: <phrase> :}
                          ) -> {}
    uq_phrase          <- <uq_atom> (%s+ <uq_atom>)*
    uq_atom            <- [^<>,; ]+
    phrase            <- <word> ([%s.]+ <word>)* /  <quoted_string>
    word              <- <atom> ! <domain_addr>
    atom              <- [^] %c()<>@,;:\".[]+
    quoted_string     <- '"' ([^"\%nl] / "\" .)*  '"'
    addr              <- <addr_spec> / "<" <addr_spec> ">"
    addr_spec         <- {:addr: <addr_chars> <domain_addr> :}
    domain_addr       <- "@" <addr_chars>
    addr_chars        <- [_%a%d][-._%a%d]*
  ]=]
```

**Comment by rebelot** [comment]
That yellowish background has to do with lsp reference highlight.

I think you can get around that within the cursorhold auto command that triggers reference highlight by first checking if you are within a string using treesitter

**Comment by indika-dev** [comment]
I use sumneko_lua and LunarVim and I found nothing regarding your idea. But your thought, that it has to do with lsp reference highlight and again reading how to configure kanagawa theme ("values in overriden will be merged [and not substituted...]") led me to the correct value in kanagawa! My solution to get rid of highlighting String Literals is:
```
require("kanagawa").setup {
    overrides = {
          LspReferenceText = { fg = "none", bg = "none" },
    },
}
```

**Comment by rebelot** [comment]
Your solution is sub optimal, and you're basically removing lsp reference highlights for everything not just strings. If do not want lsp reference highlight you should look into the on_attach function of lsp configuration or delete the corresponding autocommand

**Comment by indika-dev** [comment]
I took a look, but I found nothing. There's a .on_attach_callback and I found something [here ](https://sbulav.github.io/til/til-neovim-highlight-references/)about configuring lsp for document highlighting, but I have no idea, what the solutuin will be. You're right, that my previous solution removes lsp reference highlights completely is a bit overpowered. Therefore, this is my solution now:
```
     overrides = {
          LspReferenceText = { fg = "none", bg = "none" },
          LspReferenceRead = { bg = "#49443C" },
          LspReferenceWrite = { link = "LspReferenceRead" },
     },
```
So, normal Strings are still highlighted and references, but "my" Literal String not.
